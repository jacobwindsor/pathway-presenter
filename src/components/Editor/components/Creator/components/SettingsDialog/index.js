import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import RaisedButton from 'material-ui/RaisedButton';
import TextField from 'material-ui/TextField';
import { red500 } from 'material-ui/styles/colors';
import DeleteConfirmDialog from './components/DeleteConfirmDialog';

class SettingsDialog extends Component {
  constructor(props) {
    super(props);

    this.state = {
      title: props.title || '',
      authorName: props.authorName || '',
      wpId: props.wpId,
      version: props.version,
      deleteConfirmDialogOpen: false
    };
  }

  handleChange = targetName => e =>
    this.setState({ [targetName]: e.target.value });
  handleTitleChange = this.handleChange('title');
  handleAuthorNameChange = this.handleChange('authorName');
  handleWpIdChange = this.handleChange('wpId');
  handleVersionChange = this.handleChange('version');

  render() {
    const { handleClose, handleSave, isOpen, handleDelete } = this.props;
    const { title, authorName, wpId, version } = this.state;

    const actions = [
      <FlatButton
        label="Cancel"
        id="settings-cancel-button"
        onTouchTap={handleClose}
      />,
      <FlatButton
        id="settings-save-button"
        label="Save"
        primary={true}
        keyboardFocused={true}
        onTouchTap={() => {
          handleSave(this.state);
          handleClose();
        }}
      />
    ];

    return (
      <Dialog
        title="Settings"
        actions={actions}
        open={isOpen}
        onRequestClose={handleClose}
        autoScrollBodyContent={true}
        modal={false}
      >
        <DeleteConfirmDialog
          handleDelete={handleDelete}
          handleClose={() => this.setState({ deleteConfirmDialogOpen: false })}
          isOpen={this.state.deleteConfirmDialogOpen}
        />
        <TextField
          hintText="The TCA Cycle"
          floatingLabelText="Title"
          onChange={this.handleTitleChange}
          value={title}
          fullWidth={true}
        />
        <TextField
          hintText="Gregor Mendel, Frederick Sanger"
          floatingLabelText="Author Name(s)"
          onChange={this.handleAuthorNameChange}
          value={authorName}
          fullWidth={true}
        />
        <TextField
          hintText="WP78"
          floatingLabelText="WikiPathways ID"
          onChange={this.handleWpIdChange}
          value={wpId}
          fullWidth={true}
        />
        <TextField
          hintText="Type '0' for latest"
          floatingLabelText="WikiPathways Version"
          onChange={this.handleVersionChange}
          value={version}
          fullWidth={true}
        />
        <RaisedButton
          label="Delete"
          fullWidth={true}
          backgroundColor={red500}
          labelColor={'#ffffff'}
          onTouchTap={() => this.setState({ deleteConfirmDialogOpen: true })}
        />
      </Dialog>
    );
  }
}

SettingsDialog.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  handleSave: PropTypes.func.isRequired,
  title: PropTypes.string,
  authorName: PropTypes.string,
  version: PropTypes.number.isRequired,
  wpId: PropTypes.string.isRequired,
  handleDelete: PropTypes.func.isRequired
};

export default SettingsDialog;
