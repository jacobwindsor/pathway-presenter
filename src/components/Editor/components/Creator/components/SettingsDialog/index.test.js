import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import SettingsDialog from './index';

describe('The SettingsDialog. Allows to edit global settings that affect all slides', () => {
  it('should render correctly', () => {
    const comp = shallow(
      <SettingsDialog
        isOpen
        handleClose={() => null}
        handleSave={() => null}
        version={0}
        wpId="WP4"
        handleDelete={() => null}
        authorName="Miles Davis"
      />
    );

    expect(toJson(comp)).toMatchSnapshot();
  });
});
