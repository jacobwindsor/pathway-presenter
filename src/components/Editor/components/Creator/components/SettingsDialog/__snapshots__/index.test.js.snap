// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SettingsDialog. Allows to edit global settings that affect all slides should render correctly 1`] = `
<Dialog
  actions={
    Array [
      <FlatButton
        disabled={false}
        fullWidth={false}
        id="settings-cancel-button"
        label="Cancel"
        labelPosition="after"
        labelStyle={Object {}}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        primary={false}
        secondary={false}
    />,
      <FlatButton
        disabled={false}
        fullWidth={false}
        id="settings-save-button"
        keyboardFocused={true}
        label="Save"
        labelPosition="after"
        labelStyle={Object {}}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        primary={true}
        secondary={false}
    />,
    ]
  }
  autoDetectWindowHeight={true}
  autoScrollBodyContent={true}
  modal={false}
  onRequestClose={[Function]}
  open={true}
  repositionOnUpdate={true}
  title="Settings"
>
  <DeleteConfirmDialog
    handleClose={[Function]}
    handleDelete={[Function]}
    isOpen={false}
  />
  <TextField
    disabled={false}
    floatingLabelFixed={false}
    floatingLabelText="Title"
    fullWidth={true}
    hintText="The TCA Cycle"
    multiLine={false}
    onChange={[Function]}
    rows={1}
    type="text"
    underlineShow={true}
    value=""
  />
  <TextField
    disabled={false}
    floatingLabelFixed={false}
    floatingLabelText="Author Name(s)"
    fullWidth={true}
    hintText="Gregor Mendel, Frederick Sanger"
    multiLine={false}
    onChange={[Function]}
    rows={1}
    type="text"
    underlineShow={true}
    value="Miles Davis"
  />
  <TextField
    disabled={false}
    floatingLabelFixed={false}
    floatingLabelText="WikiPathways ID"
    fullWidth={true}
    hintText="WP78"
    multiLine={false}
    onChange={[Function]}
    rows={1}
    type="text"
    underlineShow={true}
    value="WP4"
  />
  <TextField
    disabled={false}
    floatingLabelFixed={false}
    floatingLabelText="WikiPathways Version"
    fullWidth={true}
    hintText="Type '0' for latest"
    multiLine={false}
    onChange={[Function]}
    rows={1}
    type="text"
    underlineShow={true}
    value={0}
  />
  <RaisedButton
    backgroundColor="#f44336"
    disabled={false}
    fullWidth={true}
    label="Delete"
    labelColor="#ffffff"
    labelPosition="after"
    onTouchTap={[Function]}
    primary={false}
    secondary={false}
  />
</Dialog>
`;
